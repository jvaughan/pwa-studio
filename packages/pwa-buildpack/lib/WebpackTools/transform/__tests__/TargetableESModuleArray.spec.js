const TargetableESModuleArray = require('../TargetableESModuleArray');
const ESMCollectionLoader = require('../../loaders/export-esm-collection-loader');
const testWebpackLoader = require('../../../TestHelpers/testWebpackLoader');

test('builds a module which exports an array of its imports', async () => {
    const moduleArray = new TargetableESModuleArray(
        './module-array.js',
        () => {}
    );
    moduleArray.add('import first from "first"', '{ second } from "second";');
    moduleArray.append('{ third as second } from "third"');
    moduleArray.push('{ fourth as third } from "fourth"');
    moduleArray.prepend('* as first from "elsewhere"');
    moduleArray.unshift('{ second as silver } from "second"');
    const transform = moduleArray.flush();
    expect(transform).toHaveLength(1);
    expect(transform[0]).toMatchSnapshot();
    const { output } = await testWebpackLoader.runLoader(
        ESMCollectionLoader,
        'export default [];',
        {
            query: [transform[0].options]
        }
    );
    expect(output).toMatchSnapshot();
});
