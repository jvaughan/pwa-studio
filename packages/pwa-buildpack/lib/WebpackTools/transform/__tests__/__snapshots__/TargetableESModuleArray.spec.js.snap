// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds a module which exports an array of its imports 1`] = `
Object {
  "fileToTransform": "./module-array.js",
  "options": Object {
    "items": Array [
      Array [
        "import * as first2 from \\"elsewhere\\";",
        "first2",
      ],
      Array [
        "import first from \\"first\\";",
        "first",
      ],
      Array [
        "import { second } from \\"second\\";",
        "second",
      ],
      Array [
        "import { third as second1 } from \\"third\\";",
        "second1",
      ],
      Array [
        "import { fourth as third } from \\"fourth\\";",
        "third",
      ],
    ],
    "warnings": Array [],
  },
  "transformModule": "@magento/pwa-buildpack/lib/WebpackTools/loaders/export-esm-collection-loader",
  "type": "source",
}
`;

exports[`builds a module which exports an array of its imports 2`] = `
"// generated by export-esm-collection-loader
import * as first2 from \\"elsewhere\\";
import first from \\"first\\";
import { second } from \\"second\\";
import { third as second1 } from \\"third\\";
import { fourth as third } from \\"fourth\\";

export default [
  first2,
  first,
  second,
  second1,
  third
];"
`;
